### PR TITLE
Fix win release ext

### LINKED ISSUE #7298: Cabal 3.4.0.0 release on Windows lacks extensions
**Describe the bug**

As discussed in https://github.com/Mistuke/CabalChoco/issues/3 ,the binary while it doesn't need an extension for command-line execution, it does break the platform convention and is problematic in a lot of GUI based tools.  a lot of tools filter the program type based on extensions.

**To Reproduce**

Download and unpack https://downloads.haskell.org/~cabal/cabal-install-3.4.0.0/cabal-install-3.4.0.0-x86_64-windows.zip

the file in there is `cabal` and not `cabal.exe`

**Expected behavior**
I expect file to be `cabal.exe`

**System information**
 - Windows (all)
 -  cabal: 3.4.0.0

**Additional context**

Currently blocking packaging of cabal on chocolatey and haskell-ci

https://github.com/haskell/actions/pull/39

### PR DESCRIPTION
**Pull request by jneira**
* Closes #7298 
* Being checked in gitlab here: https://gitlab.haskell.org/jneira/cabal/-/jobs/1072546

### CODE REVIEW

**Comment by jneira** [comment]
* It looks good: https://gitlab.haskell.org/jneira/cabal/-/jobs/1072546/artifacts/raw/out/cabal-install-3.7-x86_64-windows.zip

**Review by Mikolaj — APPROVED** [review]
I confirm in the test pipeline windows has .exe and at least one of the linux archives has no .exe.

Thank you so much @jneira.

File: .gitlab/ci.sh [hunk 1]
```diff
@@ -61,18 +63,18 @@ args=(
...
+cp "$(cabal list-bin ${args[@]} cabal-install:exe:cabal)" "$CI_PROJECT_DIR/out/cabal$EXE_EXT"
 cp dist-newstyle/cache/plan.json "$CI_PROJECT_DIR/out/plan.json"
 cd "$CI_PROJECT_DIR/out/"
 
 # create tarball/zip
 TARBALL_PREFIX="cabal-install-$("$CI_PROJECT_DIR/out/cabal" --numeric-version)"
 case "${TARBALL_EXT}" in
     zip)
-        zip "${TARBALL_PREFIX}-${TARBALL_ARCHIVE_SUFFIX}.${TARBALL_EXT}" cabal plan.json
+        zip "${TARBALL_PREFIX}-${TARBALL_ARCHIVE_SUFFIX}.${TARBALL_EXT}" "cabal$EXE_EXT" plan.json
```
Comment (robx) [inline .gitlab/ci.sh:74]: (I'd prefer braced variable references (`${EXE_EXT}`), if just for consistency.)

**Review by robx — APPROVED** [review]
LGTM.

File: .gitlab/ci.sh (hunk 1 above)
Comment (jneira) [reply .gitlab/ci.sh:74]: yeah, agree

Comment (jneira) [reply .gitlab/ci.sh]:
```suggestion
        zip "${TARBALL_PREFIX}-${TARBALL_ARCHIVE_SUFFIX}.${TARBALL_EXT}" "cabal${EXE_EXT}" plan.json
```

File: .gitlab/ci.sh [hunk 2]
```diff
@@ -61,18 +63,18 @@ args=(
 run cabal v2-build ${args[@]} cabal-install
 
 mkdir "$CI_PROJECT_DIR/out"
-cp "$(cabal list-bin ${args[@]} cabal-install:exe:cabal)" "$CI_PROJECT_DIR/out/cabal"
+cp "$(cabal list-bin ${args[@]} cabal-install:exe:cabal)" "$CI_PROJECT_DIR/out/cabal$EXE_EXT"
```
Comment (jneira) [inline .gitlab/ci.sh:66]: here there is no curly braces so i will not add them neither

File: .gitlab/ci.sh [hunk 3]
```diff
@@ -61,18 +63,18 @@ args=(
...
 # create tarball/zip
 TARBALL_PREFIX="cabal-install-$("$CI_PROJECT_DIR/out/cabal" --numeric-version)"
 case "${TARBALL_EXT}" in
     zip)
-        zip "${TARBALL_PREFIX}-${TARBALL_ARCHIVE_SUFFIX}.${TARBALL_EXT}" cabal plan.json
+        zip "${TARBALL_PREFIX}-${TARBALL_ARCHIVE_SUFFIX}.${TARBALL_EXT}" "cabal${EXE_EXT}" plan.json
         ;;
     tar.xz)
-        tar caf "${TARBALL_PREFIX}-${TARBALL_ARCHIVE_SUFFIX}.${TARBALL_EXT}" cabal plan.json
+        tar caf "${TARBALL_PREFIX}-${TARBALL_ARCHIVE_SUFFIX}.${TARBALL_EXT}" "cabal$EXE_EXT" plan.json
```
Comment (jneira) [inline .gitlab/ci.sh:77]:
```suggestion
        tar caf "${TARBALL_PREFIX}-${TARBALL_ARCHIVE_SUFFIX}.${TARBALL_EXT}" "cabal${EXE_EXT}" plan.json
```

**Comment by Mikolaj** [comment]
@mergifyIO backport 3.8

**Pull request merged**